grails-app/services/com/unifina/service/DataUnionJoinRequestService.groovy

```diff
@@ -38,9 +39,14 @@ class DataUnionJoinRequestService {
 
 	protected Set<Stream> findStreams(DataUnionJoinRequest c) {
 		log.debug(String.format("entering findStreams(%s)", c))
-		List<Product> products = Product.withCriteria {
+		List<Product> products = Product.createCriteria().list {
 			eq("type", Product.Type.DATAUNION)
-			eq("beneficiaryAddress", c.contractAddress)
+			if (Environment.current == Environment.TEST) {
```

Comment:
Hmm what's this? Shouldn't write environment-specific code. Does the `ilike` approach work all environments? If yes, why not just use that?

Reply:
@hpihkala `ilike` leaves a door open for SQL injection

Reply:
No it doesn't. It uses positional parameters inside.